packages/twenty-front/src/modules/workflow/search-variables/utils/variableTag.ts

```diff
@@ -0,0 +1,63 @@
+import { Node } from '@tiptap/core';
+import { mergeAttributes, RawCommands } from '@tiptap/react';
+
+declare module '@tiptap/core' {
+  interface Commands<ReturnType> {
+    variableTag: {
+      insertVariableTag: (variable: string) => ReturnType;
+    };
+  }
+}
+
+export const VariableTag = Node.create({
+  name: 'variableTag',
+  group: 'inline',
+  inline: true,
+  atom: true,
+
+  addAttributes: () => ({
+    variable: {
+      default: null,
+      parseHTML: (element) => element.getAttribute('data-variable'),
+      renderHTML: (attributes) => {
+        return {
+          'data-variable': attributes.variable,
+        };
+      },
+    },
+  }),
+
+  renderHTML: ({ node, HTMLAttributes }) => {
+    const variable = node.attrs.variable as string;
+    const variableWithoutBrackets = variable.replace(
+      /\{\{([^}]+)\}\}/g,
+      (_, variable) => {
+        return variable;
+      },
+    );
+
+    const parts = variableWithoutBrackets.split('.');
+    const displayText = parts[parts.length - 1];
+
+    return [
+      'span',
+      mergeAttributes(HTMLAttributes, {
+        'data-type': 'variableTag',
+        class: 'variable-tag',
+      }),
+      displayText,
+    ];
+  },
+
+  // @ts-expect-error - addCommands is missing from the NodeConfig type
+  addCommands: () => ({
+    insertVariableTag:
+      (variable: string) =>
+      ({ commands }: { commands: Partial<RawCommands> }) => {
+        return commands.insertContent?.({
+          type: 'variableTag',
+          attrs: { variable },
+        });
+      },
+  }),
```

Comment:
I think you can make TypeScript happy with the following code. It seems to work still. Does it make sense?

```suggestion
    addCommands: () => ({
      insertVariableTag:
        (variable: string) =>
        ({ commands }) => {
          commands.insertContent({
            type: 'variableTag',
            attrs: { variable },
          });

          return true;
        },
    }),
```

Reply:
🤯 I did try the boolean but I still using `: { commands: Partial<RawCommands> }` as type. Thank you!

Reply:
Haha 😄 I found somewhere in the documentation where they added custom commands and used this syntax.